src/libraries/System.Net.Requests/src/System/Net/HttpWebResponse.cs

```diff
@@ -371,5 +380,53 @@ private void CheckDisposed()
         }
 
         private static string GetHeaderValueAsString(IEnumerable<string> values) => string.Join(", ", values);
+
+        internal sealed class TruncatedReadStream(Stream innerStream, int maxSize) : Stream
+        {
+            public override bool CanRead => true;
+
+            public override bool CanSeek => false;
+
+            public override bool CanWrite => false;
+
+            public override long Length => throw new NotSupportedException();
+
+            public override long Position { get => throw new NotSupportedException(); set => throw new NotSupportedException(); }
+
+            public override void Flush() => innerStream.Flush();
+            public override Task FlushAsync(CancellationToken cancellationToken) => base.FlushAsync(cancellationToken);
```

Comment:
Shouldn't this also delegate to `innerStream`?

```suggestion
            public override Task FlushAsync(CancellationToken cancellationToken) => innerStream.FlushAsync(cancellationToken);
```